server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/LocalShardsBalancer.java

```diff
@@ -959,6 +979,11 @@ AllocateUnassignedDecision decideAllocateUnassigned(final ShardRouting shard) {
                 nodeDecisions.add(new NodeAllocationResult(current.getNode(), current.getCanAllocateDecision(), ++weightRanking));
             }
         }
+
+        if (preferRandomShardAllocation && !minNodes.isEmpty()) {
+            minNode = minNodes.get(Randomness.get().nextInt(minNodes.size()));
```

Comment:
lets check that there is no additional cost of creating new Random object within Randomness class.